### ISSUE TITLE
Comment editing

### ISSUE THREAD

**Issue by torgeirl** [devilry/devilry-django#1016]
Currently there's no way for an examiner to edit typos after a feedback has been provided.

There's some big issues with allowing anything to be edited anytime, but if we find a right balance this is something you would want in the Slack era.

**Possible solution**

- Allow examiners to edit their feedbacks (and comments, if it's the exact same data model)
- Give a visual indication that it has been edited and when (ie «edited Wedneday April 11, 2018 at 10:29»)
- Allow users that can see that post to toggle the editions via a fall-down list
- Bonus: editions highlights changes similar to how Github shows changes to a text file

**Comment by espenak** [comment]
This should be generalized into comment editing. We should have the same history when students or admins change comments that have already been "published".

Instead of a falldown list, I think we should put view comment history into a separate view. So you just click on the _edited Wedneday April 11, 2018 at 10:29_ text (a link) to go to a view where you can see the history for the comment (just like viewing a commit on github).

I think the first version should just be a listing of all the changes with a two column layout with _changed from_ in the left column and _changed to_ in the right column, and the datetime of the change. This means that the right column will be repeated on the next line if there are multiple changes. On small displays, we should just show the changes in an ordered list (in practice this means that we just remove the "duplicates" in the left column on line 2 and onwards via CSS). We can look at showing a diff later on. Python has built-in text diffing support, but that would mean showing the diff in the raw markdown of the comments, not in the rendered HTML (which is much harder).

Showing the percentage of the change should also be postponed since it requires diffing to be correct. Consider this change with a _dumb_ length diff in the change:
```
## Changed from
Hello crual world

## Changed to
Hello cruel world
```
The change in characters and words are 0%!

**Comment by espenak** [comment]
I renamed the issue to _Comment editing_. We already have _grade editing_, so the only new thing in this issue is comment editing. When we introduce this for feedback comments, there is no reason to not include it for all comments.

**Comment by torgeirl** [comment]
@espenak: your suggestion sounds good. :+1: This is something our examiners ask for so I hope we'll be able to include at least the simplified version this Spring.

Is the simple version something where we'll be able to reuse existing resources, and therefore something we can put into limited production somewhat soon?

**Comment by espenak** [comment]
This should be fairly straight forward to implement, so I think we should be able to get it out fairly soon after the 3.2 release is ready. We will just have to discuss what to include in a 3.3 release, and weight this feature against other new features that may come up before we start planning 3.3.
